NuRadioReco/modules/base/module.py

```diff
@@ -0,0 +1,57 @@
+from functools import wraps
+from timeit import default_timer as timer
+import NuRadioReco.framework.event
+import NuRadioReco.framework.base_station
+
+
+def register_run(level=None):
+    if level not in ["station", "event"]:
+        raise NotImplementedError("The level needs to be either 'station' or 'event'")
+
+    def run_decorator(run):
+
+        @wraps(run)
+        def register_run_method(self, *args, **kwargs):
```

Comment:
Please document these. They are hard to understand for those, who were not involved in making these. If we want to keep it sustainable, we need to document our choices.